integration_tests/CMakeLists.txt

```
@@ -688,14 +688,14 @@ RUN(NAME intrinsics_138 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) #
 RUN(NAME intrinsics_139 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # modulo
 RUN(NAME intrinsics_140 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # selected_int_kind
 RUN(NAME intrinsics_141 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # scale
-RUN(NAME intrinsics_142 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dot_product
+RUN(NAME intrinsics_142 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # dot_product
 RUN(NAME intrinsics_143 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dot_product
-RUN(NAME intrinsics_144 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dot_product
+RUN(NAME intrinsics_144 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # dot_product
 RUN(NAME intrinsics_145 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dot_product
 RUN(NAME intrinsics_146 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dprod
 RUN(NAME intrinsics_147 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # pack
 RUN(NAME intrinsics_148 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # pack
-RUN(NAME intrinsics_149 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc NOFAST) # unpack
+RUN(NAME intrinsics_149 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc NOFAST fortran) # unpack
```

I think this does not get considered as a part of LABELS at the moment.

```suggestion
RUN(NAME intrinsics_149 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran NOFAST) # unpack
```

---

What was the issue here? Is it the sequence of labels?

---

Can you check if `fortran` is considered as a part of `LABELS` with your current change?

---

Oh I see
When I run the following command:
```console
$ ./integration_tests/run_tests.py -b fortran
```
A number of tests run but intrinsics_149 is not present there.

---

And it fails after applying the suggestion.
```console
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:55:132:

   55 | print *, (unpack([23.12000084, 23.12000084, 23.12000084, 23.12000084], [1, 0, 1, 0], [1.00000000, 2.00000000, 3.00000000, 4.00000000]))
      |                                                                                                                                    1
Error: Line truncated at (1) [-Werror=line-truncation]
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:55:132:

   55 | print *, (unpack([23.12000084, 23.12000084, 23.12000084, 23.12000084], [1, 0, 1, 0], [1.00000000, 2.00000000, 3.00000000, 4.00000000]))
      |                                                                                                                                    1
Error: Syntax error in array constructor at (1)
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:56:132:

   56 | unity__ = (unpack([23.12000084, 23.12000084, 23.12000084, 23.12000084], [1, 0, 1, 0], [1.00000000, 2.00000000, 3.00000000, 4.00000000]))
      |                                                                                                                                    1
Error: Line truncated at (1) [-Werror=line-truncation]
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:56:132:

   56 | unity__ = (unpack([23.12000084, 23.12000084, 23.12000084, 23.12000084], [1, 0, 1, 0], [1.00000000, 2.00000000, 3.00000000, 4.00000000]))
      |                                                                                                                                    1
Error: Syntax error in array constructor at (1)
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:25:7:

   25 | mask = reshape([1, 0, 0, 1], shape(mask))
      |       1
Warning: Extension: Conversion from INTEGER(4) to LOGICAL(4) at (1)
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:39:9:

   39 | mask_r = reshape([1, 0, 0, 1, 1, 0], shape(mask_r))
      |         1
Warning: Extension: Conversion from INTEGER(4) to LOGICAL(4) at (1)
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:63:8:

   63 | mask_ = [1, 0, 1, 0]
      |        1
Warning: Extension: Conversion from INTEGER(4) to LOGICAL(4) at (1)
CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90:72:9:

   72 | mask_c = reshape([1, 0, 1, 0, 1, 0, 1, 0], shape(mask_c))
      |         1
Warning: Extension: Conversion from INTEGER(4) to LOGICAL(4) at (1)
f951: some warnings being treated as errors
The command 'gfortran -fno-backtrace -o CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o -c CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o.tmp.f90' failed.
CMakeFiles/intrinsics_200.dir/intrinsics_200.f90.o.tmp.f90:6:7:

    6 | mask = [1, 0, 1, 0, 1]
      |       1
Warning: Extension: Conversion from INTEGER(4) to LOGICAL(4) at (1)
make[2]: *** [CMakeFiles/intrinsics_149.dir/build.make:75: CMakeFiles/intrinsics_149.dir/intrinsics_149.f90.o] Error 11
make[1]: *** [CMakeFiles/Makefile2:1983: CMakeFiles/intrinsics_149.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 60%] Linking Fortran executable intrinsics_174
[ 60%] Linking Fortran executable intrinsics_199
[ 60%] Linking Fortran executable intrinsics_175
[ 61%] Linking Fortran executable intrinsics_206
[ 61%] Linking Fortran executable intrinsics_200
[ 61%] Linking Fortran executable modules_06
[ 61%] Linking Fortran executable modules_01
[ 61%] Built target intrinsics_175
[ 61%] Built target intrinsics_174
[ 61%] Built target intrinsics_206
[ 61%] Built target intrinsics_199
[ 61%] Built target intrinsics_200
[ 61%] Built target modules_06
[ 61%] Built target modules_01
make: *** [Makefile:101: all] Error 2
Command failed.
```

---

Good, it means:

- `fortran` is not being considered as part of LABELS. Hence my suggestion above.
- this specific test case is not/yet-to-be supported by the `fortran` backend.

---

Got it. Thanks for this, really helpful :)